File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/goods/Goods.java
package org.morecup.jimmerddd.java.sample.domain.goods;

import org.babyfish.jimmer.sql.Entity;
import org.morecup.jimmerddd.core.aggregateproxy.AggregateProxy;
import org.morecup.jimmerddd.core.event.EventHandler;
import org.morecup.jimmerddd.java.sample.domain.base.BaseEntity;
import org.morecup.jimmerddd.java.sample.domain.base.DomainEvent;


@Entity
public interface Goods extends BaseEntity {
    String name();
    String nowAddress();
}

File: jimmer-ddd-kotlin-sample/domain/src/main/kotlin/org/morecup/jimmerddd/kotlin/sample/domain/goods/Goods.kt
package org.morecup.jimmerddd.kotlin.sample.domain.goods

import org.babyfish.jimmer.sql.Entity
import org.morecup.jimmerddd.core.event.publishEvent
import org.morecup.jimmerddd.kotlin.aggregateproxy.KAggregateProxy
import org.morecup.jimmerddd.kotlin.sample.domain.base.BaseEntity

@Entity
interface Goods : BaseEntity {
    val name: String
    val nowAddress: String
}
val goodsAggregateProxy = KAggregateProxy(GoodsDraft::class)
class GoodsImpl(val goods: GoodsDraft) : GoodsDraft by goods {
    fun rename(newName: String) {
        goods.publishEvent(1)
        this.name = newName
    }
}